app/models/amusement_park.rb

```diff
@@ -1,3 +1,8 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+
+  def unq_mechs
+    # self.joins(:ride_mechanics)
```

Comment:
The easier way to do this would actually be to add additional associations! Since the association with `rides` is already available, you can add `has_many :mechanics, through: :rides` in order to be able to do `self.mechanics.distinct` to get the list of unique mechanics.

Reply:
Harder way: `self.rides.joins(:mechanics).distinct`